pkg/keyspace/tso_keyspace_group.go

```diff
@@ -352,6 +355,27 @@ func (m *GroupManager) saveKeyspaceGroups(keyspaceGroups []*endpoint.KeyspaceGro
 	})
 }
 
+// saveGroupWithUpdateKeyspace will try to save the given keyspace groups into the storage.
+// It only update the keyspace field for the keyspace group.
+func (m *GroupManager) saveGroupWithUpdateKeyspace(keyspaceGroup *endpoint.KeyspaceGroup) error {
```

Comment:
If we only run it on the leader node, do we still need it?

Reply:
I think it is still necessary because we provide an external interface to set node for keyspace groups.

Reply:
But it can be only executed by the leader?

Reply:
Yes, but `UpdateKeyspaceForGroup` is called by manager of keyspace. 

For example, when manager of keyspace call `UpdateKeyspaceForGroup`, but we exec `set node` before `updateKeyspaceForGroupLocked`, which is similiar with [failpoint.Inject](https://github.com/tikv/pd/blob/fe58ed103afc060b97f6884c71085d78b12fe5bc/pkg/keyspace/tso_keyspace_group.go#L418).

Reply:
We have a lock. Does it prevent this thing from happening?

Reply:
yes, updated, PTAL